lib/Model/Importer/Binary/PECOFFImporter.cpp

```diff
@@ -74,22 +74,48 @@ Error PECOFFImporter::import() {
     }
     const object::coff_section *CoffRef = *SecOrErr;
 
-    // VirtualSize might be larger than SizeOfRawData (extra data at the end of
-    // the section) or viceversa (data mapped in memory but not present in
-    // memory, e.g., .bss)
-    uint64_t SegmentSize = std::min(CoffRef->VirtualSize,
-                                    CoffRef->SizeOfRawData);
-
     MetaAddress Start = ImageBase + u64(CoffRef->VirtualAddress);
     Segment Segment({ Start, u64(CoffRef->VirtualSize) });
 
     Segment.StartOffset = CoffRef->PointerToRawData;
-    Segment.FileSize = Segment.StartOffset + SegmentSize;
+
+    // VirtualSize might be larger than SizeOfRawData (extra data at the end of
+    // the section) or viceversa (data mapped in memory but not present in
+    // memory, e.g., .bss)
+    Segment.FileSize = CoffRef->SizeOfRawData;
+
+    // Since it is possible that the file size if greater than VirtualSize
+    // because SizeOfRawData is rounded, but VirtSize is not,
+    // we work it around here.
+    // TODO: Revisit this.
+    if (Segment.FileSize > Segment.VirtualSize)
+      Segment.VirtualSize = Segment.FileSize;
 
     Segment.IsReadable = CoffRef->Characteristics & COFF::IMAGE_SCN_MEM_READ;
     Segment.IsWriteable = CoffRef->Characteristics & COFF::IMAGE_SCN_MEM_WRITE;
     Segment.IsExecutable = CoffRef->Characteristics
                            & COFF::IMAGE_SCN_MEM_EXECUTE;
+
+    // NOTE: Unlike ELF, PE/COFF does not have segments. Instead,
+    // it has sections only. All the raw data in a section must be loaded
+    // contiguously. Segments just map file range to virtual address space
+    // range.
+    auto SectionStart = ImageBase + u64(CoffRef->VirtualAddress);
+    uint64_t Size = u64(CoffRef->VirtualSize);
+    auto SectionEnd = SectionStart + Size;
+
+    if (SectionStart.isValid() and SectionEnd.isValid()
+        and SectionStart.addressLowerThan(SectionEnd)) {
+      model::Section NewSection(SectionStart, Size);
+      if (auto SectionName = TheBinary.getSectionName(CoffRef))
+        NewSection.Name = SectionName->str();
+      NewSection.ContainsCode = Segment.IsExecutable;
+      NewSection.verify(true);
```

Comment:
Can you wrap this in an `assert`? One day we'll want to have generic soft-fail for model validation failures and having them wrapped in an `assert` will make the refactoring a bit easier.